### PR TITLE
Remove RecordTagHelper

### PR DESCRIPTION
**Pull request by todd**
Per DHH in #18337, I've extracted `ActionView::Helpers::RecordTagHelper` to an external gem (source currently lives at [todd/record_tag_helper](https://github.com/todd/record_tag_helper)). I also added extraction notices for anyone upgrading who was using these methods - the corresponding code and tests should probably be fully removed at the next minor release after 5.0.0.

Regarding the gem, I still need to write some documentation (the README is kinda sparse at the moment) and I'd appreciate someone else going over the code to make sure I'm not missing anything, though it seems pretty straightforward. There's also the question of whether the source for `record_tag_helper` should stay under my own account or get forked into the Rails org - I don't have a strong opinion either way. The answer to that question should also dictate who publishes the gem, so I'm holding off on pushing anything to RubyGems.

### CODE REVIEW

File: actionview/test/template/record_tag_helper_test.rb [hunk 1]
```diff
@@ -24,94 +24,52 @@ def setup
   end
 
   def test_content_tag_for
-    expected = %(<li class="record_tag_post" id="record_tag_post_45"></li>)
-    actual = content_tag_for(:li, @post)
-    assert_dom_equal expected, actual
+    assert_raises(NoMethodError) { content_tag_for(:li, @post) }
   end
```
Comment (carlosantoniodasilva) [inline actionview/test/template/record_tag_helper_test.rb]: I think you can keep only 1 test for each method, should be enough.

File: actionview/CHANGELOG.md [hunk 2]
```diff
@@ -28,5 +28,9 @@
 
     *Angelo Capilleri*
 
+*   Extracted `ActionView::Helpers::RecordTagHelper` to external gem
+    (`record_tag_helper`) and added removal notices.
+
+    *Todd Bealmear*
```
Comment (carlosantoniodasilva) [inline actionview/CHANGELOG.md]: This should be moved to the top.

File: actionview/test/template/record_tag_helper_test.rb (hunk 1 above)
Comment (todd) [inline actionview/test/template/record_tag_helper_test.rb]: Yeah, I figured that would probably be enough, but I wanted to cover all the cases. I'll slim this down.

File: actionview/CHANGELOG.md (hunk 2 above)
Comment (todd) [inline actionview/CHANGELOG.md]: Wasn't sure about the ordering, I'll bump it up.

**Comment by egilburg** [comment]
Public API changes should go through deprecation process

**Comment by carlosantoniodasilva** [comment]
@egilburg please read the discussion on the related issue, linked above. It's going to be extracted to a gem.

**Comment by egilburg** [comment]
@carlosantoniodasilva I did read about the gem extraction. In such cases, do we expect users to manually notice their app is broken (e.g. first time they get a runtime error) and add the gem entry, or would they get some runtime warnings first? 

I also do realize we have changelogs and upgrade instructions. But I didn't know there is a deprecation difference between code which requires a gem change from code that requires any other equally simple change (e.g. change method call signature somewhere in the app)

**Comment by carlosantoniodasilva** [comment]
The functionality will still be available for anyone who wants to use it. I don't believe people will just upgrade to Rails 5 and put their apps in production right away to see it blowing up. In short, this is pretty much the same we did with respond_to/respond_with and the responders gem, which is totally fine as long as there's an upgrade path imo.

**Comment by robertomiranda** [comment]
[record_tag_helper](https://github.com/todd/record_tag_helper) should be live under /rails?

**Comment by carlosantoniodasilva** [comment]
[We will have it under rails for now](https://github.com/rails/rails/issues/18337#issuecomment-69512848), as we generally do with other extractions at the beginning.

**Comment by rafaelfranca** [comment]
@todd could you transfer the repository to rails organization? I'll make sure you will have full access to it.

**Comment by todd** [comment]
@rafaelfranca I'm not an admin for the Rails org, so I can't transfer ownership to Rails. Can I add you as an admin on the repo and have you transfer ownership?

**Comment by rafaelfranca** [comment]
@todd :+1:

**Comment by todd** [comment]
@rafaelfranca I added you as a collaborator, but I'm not sure if that grants you the ability to transfer repo ownership. It doesn't appear there's a way for me to edit your permissions on the repo beyond that. Let me know if we need to figure something else out.

**Comment by rafaelfranca** [comment]
@todd :sweat_smile: I think you will have to transfer for me and so I can transfer for rails.

**Comment by todd** [comment]
@rafaelfranca Sheesh. Alright, done. :stuck_out_tongue:

**Comment by rafaelfranca** [comment]
You should have access to the gem. Thank you so much for working on this @todd

**Comment by todd** [comment]
@rafaelfranca My pleasure. What should the strategy/process look like for actually releasing the gem?

**Comment by rafaelfranca** [comment]
We usually release it with the beta release of Rails.

**Comment by todd** [comment]
Ok, should I be responsible for that or will someone from the Core Team take the lead? Either way is fine, just trying to scope out my role moving forward.

**Comment by rafaelfranca** [comment]
@todd I think for the first release it is better to us handle, but you will have access to push the gem too.

**Comment by todd** [comment]
:+1: Sounds good.

**Comment by claudiob** [comment]
Since this is now out of Rails, it would be great to have a running version of the gem on RubyGems as soon as possible, for people who want to transition.

To avoid squatting the `record_tag_helper` name, I *blocked* it on RubyGems by pushing a "0.0.0" version. All the common gem owners of Rails project are now owners of the `record_tag_helper` gem as well, so you can push a new version as soon as it's ready.

**Pull request merged**